src/Microsoft.TemplateEngine.Edge/Settings/SettingsLoader.cs

```diff
@@ -295,6 +332,21 @@ private void EnsureLoaded()
             }
 
             _isLoaded = true;
+
+            //first run execution
```

Comment:
There is one big difference between this and original logic... In original logic, if file did not exist, whole "BaseDir" folder was deleted, that is useful if first run(creation of settings.json and other files) is interrupted and they are now corrupt or in invalid state... So as long as first run did not execute successfully, it keeps deleting folder and trying...
Which makes me wonder if we re-add that delete logic... Should this check move to beginning of this method to delete folder asap...

Reply:
I'll switch to old logic here, and have cookie file only written when first run is successful - i think it's better this way.
I'm not sure if we should throw exception if first run throws, maybe we should; otherwise host might miss it.

I'm not sure if we should kick the reset whole settings on first run for CLI: the only operation remaining is below:

```
environmentSettings.SettingsLoader.Components.RegisterMany(typeof(New3Command).GetTypeInfo().Assembly.GetTypes());
```

Reply:
If first run is moved to beginning of method, then components won't be initialized to run `RegisterType`, but this can be solved.